truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java

```diff
@@ -1296,13 +1296,17 @@ public Object deleteAtNullOrEmpty(RubyArray array, int index) {
         private final BranchProfile nextProfile = BranchProfile.create();
         private final BranchProfile redoProfile = BranchProfile.create();
 
+        private final RubySymbol eachSymbol;
+
         public EachNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
+            eachSymbol = getContext().getSymbolTable().getSymbol("each");
```

Comment:
I think we use `getContext().newSymbol()` almost exclusively.  It might make future refactorings easier to stick with that.